llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -2482,10 +2482,21 @@ Instruction *InstCombinerImpl::foldICmpShrConstant(ICmpInst &Cmp,
   // those conditions rather than checking them. This is difficult because of
   // undef/poison (PR34838).
   if (IsAShr && Shr->hasOneUse()) {
+    if (IsExact && (Pred == CmpInst::ICMP_SLT || Pred == CmpInst::ICMP_ULT) &&
+        !C.isMinSignedValue() && (C - 1).isPowerOf2()) {
```

Comment:
Doesn't `(C - 1).isPowerOf2()` preclude `C.isMinSignedValue()`?

Reply:
Yes, I've removed that now. Thanks.